### ISSUE TITLE
Accessor functions for response header

### ISSUE THREAD

**Issue by kilianmh** [fukamachi/lack#89]
It might be convenient  to have simple (syntactic-sugar) 
functions for accessing and setf'ing single response header.

A generic one would be:
```common-lisp
(defun response-header (response header)
  (getf (response-headers response) header))

(defun (setf response-header) (value response header)
  (setf (getf (response-headers response) header)
	value))
```
Alternatively, or on top of that we could introduce access/setf 
functions for frequently used header (e.g. content-type, etc.):
```common-lisp
(defun response-content-type (response)
  (getf (response-headers response) :content-type))

(defun (setf response-content-type) (value response)
  (setf (getf (response-headers response) :content-type)
	value))
```
Do you like the idea, or unnecessary code bloat @fukamachi ?

**Comment by fukamachi** [comment]
I used to be skeptical of the need for those functions because I didn't think end users would use Lack request objects directly; however, they do.
It doesn't harm usability (people still can treat it as plist if needed), so it'd be great to have.

**Comment by kilianmh** [comment]
So then which [Standard response fields](https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Standard_response_fields) should we include? All which are permanent and not obsolete? That would be a lot though.
And which of the [Common non-standard response fields](https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Common_non-standard_response_fields)?

**Comment by fukamachi** [comment]
Response headers that web applications would use are limited.
I looked over applications I was involved in, but I found only `content-type` and `access-control-*` were set through `request-headers`.
Not limited, but adding accessors for these is sufficient for me. Of course, it's fine to add more if you need any, though.

**Comment by kilianmh** [comment]
> I looked over applications I was involved in, but I found only content-type and access-control-* were set through request-headers. 

Yeah these are the basic ones that most use. However some people might want to use more exotic standard or Common non-standard fields.

Here are some candidates from [HTTP Security Response Headers Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/HTTP_Headers_Cheat_Sheet.html#http-security-response-headers-cheat-sheet)
- X-Frame-Options
- X-XSS-Protection 
- X-Content-Type-Options
- Referrer-Policy
- Strict-Transport-Security
- Content-Security-Policy 
- Permissions-Policy
- Cross-Origin-Embedder-Policy
- Cross-Origin-Opener-Policy
- Cross-Origin-Resource-Policy
- Server

Is the prefix "response-" needed, or not?
```common-lisp
(content-type response)
```
Should we also introduce set-* functions so one less parentheses is necessary:
```common-lisp
(set-response-content-type response "application/json")
;; vs:
(setf (response-content-type response) "application/json")
```

Also a general-purpose functions to add/update multiple Headers
in one go should be useful:
```common-lisp
(set-headers response
             :content-type "application/json"
             :server "default")
```

**Comment by fukamachi** [comment]
> Yeah these are the basic ones that most use. However some people might want to use more exotic standard or Common non-standard fields.
>
> Here are some candidates from [HTTP Security Response Headers Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/HTTP_Headers_Cheat_Sheet.html#http-security-response-headers-cheat-sheet)

Yeah, I never used them, but it seems a good list to be added.

> Is the prefix "response-" needed, or not?

Since other functions in Lack.Response, it's better to have a prefix of `response-`.

> Should we also introduce set-* functions so one less parentheses is necessary:

I don't really like it, and maybe there's not much benefits.

> Also a general-purpose functions to add/update multiple Headers
in one go should be useful:

It looks neat.
However, it's ambiguous how it works if it appends or replaces, and how it treats multiple HTTP headers with the same key, I think.
